### DISCUSSION TITLE
TSC Minutes May 2022

### DISCUSSION THREAD

**Discussion by lgirdwood** [thesofproject/sof#7401] [Team Posts]
@thesofproject/steering-committee minutes from May TSC.

Present: @lgirdwood @dbaluta @cujomalainey @bzhg @mwasko @sathya-nujella @balakishorepati @yaochunhung @kuanhsuncheng @bhiregou (I think I have everyone's named mapped from the call to your github IDs - if not pls correct)

Apologies @plbossart 

**Agenda**

1) Zephyr status and proposed next steps.
Zephyr native API integration time frame is for completion of the following for SOF v2.3 at the end of Q3: DAI, DMA, IPC, IDC, PM and logger (with some logger backends upstream and merged too)
.
xtos related code will be moved to a new folder using RTOS abstractions based on Zephyr APIs. This should make RTOS abstractions cleaner and easier to integrate with.

2) Module API updates and proposed next steps
Library manager PR progressing with ability to load static libraries/modules for SOF v2.3. Dynamic libraries/modules to come after v2.3. 
Module adapter changes in progress for supporting Windows modules in v2.3 (both C++ and C modules).

3) Kernel status and proposed next steps.
Custom params for module adapter for compressed params in review.
Open on aligning ABI kernel header with FW header updates. i.e.  FW gets lots of ABI updates but should these all be applied for kernel (when not used in kernel) and should kernel ABI update upstream cadence match FW (or should ABI updates be squashed into single patches for easier upstreaming). @plbossart any guidance here ?   

4) Kconfig build All option
All agreed good to have, but needs to depend on Zephyr to abstract. Needs 3rd party stubs (item 5).

5) Build-able 3rd party module wrappers using stubs.
All agreed. Needs "howto upstream a processing module" manual for upstreaming modules in sof-docs. @mwasko has documentation updates in progress. ETA for v2.3

6) CODEOWNERs updates for better reviews. 
Agreed. AR for @lgirdwood to start creating more teams.

7) SPARSE support - lots of warning being reported now for us to fix.
SPARSE initial implementation now upstream in Zephyr and SOF. Full integration with errors and warning fixed for v2.3

8) DMA based scheduling - needs better Zephyr integration with Zephyr preemption task levels.
NXP to look at as main user. AMD also user.

9) Security for fuzzing kernel IPC.
Google working on patches to fuzz kernel logic for IPCs coming from FW. Patches in progress.

10) ABI update needed to determine what's connected to AEC, smart AMP. No ability to determine what's on the other end atm, may need a away to propagate IDs to userspace via topology tags. Google to lead and present some RFCs for discussion.

**Comment by cujomalainey** [comment]
regarding #6 i have created @thesofproject/google will go and update CODEOWNERS as well
